internal/identity/github.go

```diff
@@ -0,0 +1,171 @@
+package identity
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"gopkg.in/square/go-jose.v2/jwt"
+
+	"github.com/pomerium/pomerium/internal/httputil"
+	"github.com/pomerium/pomerium/internal/sessions"
+	"github.com/pomerium/pomerium/internal/version"
+
+	"golang.org/x/oauth2"
+)
+
+const (
+	defaultGitHubProviderURL = "https://github.com"
+	githubAuthURL            = "/login/oauth/authorize"
+	githubUserURL            = "https://api.github.com/user"
+	githubUserGroupURL       = "https://api.github.com/user/orgs"
+	githubRevokeURL          = "https://github.com/oauth/revoke"
+
+	refreshDeadline = time.Minute * 60
+)
+
+// GitHubProvider is an implementation of the OAuth Provider.
+type GitHubProvider struct {
+	*Provider
+
+	// baseURL would be different in the enterprise version
```

Comment:
i'm not sure what `baseURL` refers to here anymore. looks like it may have been removed?

Reply:
Thanks for the observation, it needs to be removed, I'll work on it.